### PR TITLE
path routing with graphs

### PR DESCRIPTION
**Pull request by Asilon47**
Added path routing behavior plugin that calculates paths using graph representations instead of occupancy grids.

---

## Basic Info

| Info | Added a plugin for graph-based TSP|
| ------ | ----------- |
| Issue(s) this addresses   | N/A (New feature)  |
| ROS2 version tested on | Humble |
| Aerial platform tested on | Gazebo (Aerostack2 simulation)|

---

## Description of contribution in a few bullet points

* Introduced the `tsp` path planning behavior plugin to the `as2_behaviors_path_planning` package.
* Implemented graph-based routing algorithms to serve as an alternative to the existing occupancy grid path planners (like A* or Voronoi).
* Enables efficient pathfinding operations by routing through pre-defined graph nodes and edges rather than grid cells.

## Description of documentation updates required from your changes
* Update the `as2_behaviors_path_planning` README to include the description, launch arguments, and configuration parameters for using the new tsp graph-based plugin.

---

## Future work that may be required in bullet points

* Refactor and align variable names within the plugin to ensure strict consistency with the `aerostack2` coding guidelines.
* Add or expand unit tests specifically for graph node generation and edge-case routing scenarios.

### CODE REVIEW

**Comment by miferco97** [comment]
@GPatiA2  @Asilon47  @RPS98  what does CDTI means?

File: as2_behaviors/as2_behaviors_path_planning/plugins/cdti/include/cdti.hpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+/*!******************************************************************************
+ *  \file       cdti_routing.hpp
+ *  \brief      graph_routing header file.
+ *  \authors    Asil Arnous
+ ********************************************************************************/
+
+#ifndef CDTI_HPP_
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/cdti/include/cdti.hpp]: Remove references to CDTI. This is the name of the project this pull request is originated from. The names of the plug-in and files should reference what the plug-in does

File: as2_behaviors/as2_behaviors_path_planning/plugins/cdti/include/cdti.hpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+#include <as2_behaviors_path_planning/path_planner_plugin_base.hpp>
+
+#include "cdti_searcher.hpp"
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "geometry_msgs/msg/point_stamped.hpp"
+#include "nav_msgs/msg/occupancy_grid.hpp"
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/cdti/include/cdti.hpp]: You are not using an occupancy grid. Remove this dependency

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/tsp.hpp [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  bool enable_visualization_;
+  double penalty_x;
+  double penalty_y;
+  rclcpp::Subscription<as2_msgs::msg::AGraph>::SharedPtr occ_grid_sub_;
+
+  rclcpp::Publisher<visualization_msgs::msg::Marker>::SharedPtr viz_pub_;
+  rclcpp::Publisher<nav_msgs::msg::OccupancyGrid>::SharedPtr viz_obstacle_grid_pub_;
+
+private:
+  void occ_grid_cbk(const as2_msgs::msg::AGraph::SharedPtr msg);
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/tsp.hpp]: You should not be subscribing to an occupancy grid. Your subscriber should listen to messages of type AGraph

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/tsp_algorithm.hpp [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+#include <vector>
+
+// OpenCV
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <opencv2/opencv.hpp>
+
+// Std libraries
+
+static int convertPointToInt(const cv::Point2i & point)
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/tsp_algorithm.hpp]: This conversion should not be fixed. Your plugin works with graphs of inter connected nodes that have properties

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/tsp_algorithm.hpp [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+  double get_g_cost() const {return g_cost_;}
+  double get_h_cost() const {return h_cost_;}
+  double get_f_cost() const {return f_cost_;}
+
+  operator int() const {return convertPointToInt(coordinates_);}
+
+  double computeCosts(const cv::Point2i & goal)
+  {
+    const cv::Point2i & point = get_coordinates();
+    h_cost_ = std::sqrt(
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/tsp_algorithm.hpp]: This cost is the only one you should be calculating while creating the graph. Other costs are maintained during algorithm execution

File: as2_behaviors/as2_behaviors_path_planning/plugins/cdti/include/cdti_algorithm.hpp [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+        // g_cost_ += 1;
+        g_cost_ += 0;
+      }
+    }
+    f_cost_ = g_cost_ + h_cost_;
+    return f_cost_;
+  }
+};
+
+class CDTIPlanner
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/cdti/include/cdti_algorithm.hpp]: Remove CDTI Reference

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/tsp_algorithm.hpp [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+    }
+    f_cost_ = g_cost_ + h_cost_;
+    return f_cost_;
+  }
+};
+
+class CDTIPlanner
+{
+private:
+  cv::Mat ocuppancy_grid_;
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/tsp_algorithm.hpp]: Remove occupancy_grid_ and replace with AGraph

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/tsp_algorithm.hpp [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+  cv::Point2i goal_;
+  cv::Point2i origin_point_;
+
+  // using hash tables for speed
+  std::unordered_map<int, NodePtr> nodes_visited_;
+  std::unordered_map<int, NodePtr> nodes_to_visit_;
+  std::vector<cv::Point2i> valid_movements_;
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/tsp_algorithm.hpp]: No need for this since valid_movements_ is given by the adjacency list entry of a node

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/tsp_algorithm.hpp [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+
+    valid_movements_.emplace_back(-1, 0);
+    valid_movements_.emplace_back(0, -1);
+    valid_movements_.emplace_back(0, 1);
+    valid_movements_.emplace_back(1, 0);
+
+    valid_movements_.emplace_back(-1, -1);
+    valid_movements_.emplace_back(-1, 1);
+    valid_movements_.emplace_back(1, -1);
+    valid_movements_.emplace_back(1, 1);
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/tsp_algorithm.hpp]: No need to initialize this

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/tsp_algorithm.hpp [hunk 10]
```diff
@@ -0,0 +1,257 @@
...
+  }
+
+  void setOcuppancyGrid(const cv::Mat & mat) {ocuppancy_grid_ = mat.clone();}
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/tsp_algorithm.hpp]: Replace OccupancyGrid with AGraph

File: as2_behaviors/as2_behaviors_path_planning/plugins/cdti/include/cdti_searcher.hpp [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+ *  \authors    Asil Arnous
+ ********************************************************************************/
+
+#ifndef CDTI_SEARCHER_HPP_
+#define CDTI_SEARCHER_HPP_
+
+#include <opencv2/core/types.hpp>
+#include <opencv2/opencv.hpp>
+
+#include "nav_msgs/msg/occupancy_grid.hpp"
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/cdti/include/cdti_searcher.hpp]: Remove OpenCV dependencies

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/tsp_searcher.hpp [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+class CDTIRoutingSearcher
+{
+public:
+ std::vector<Point2i> solve_dijkstra(
+  const as2_msgs::msg::AGraph & graph, 
+    double penalty_x, double penalty_y);
+bool cell_occupied(Point2i cell);
+private:
+  int cell_to_index(int x, int y, int width) { return y * width + x; }
+  Point2i index_to_cell(int index, int width) { return {index % width, index / width}; }
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/tsp_searcher.hpp]: Conversion is given by node coordinates

File: as2_behaviors/as2_behaviors_path_planning/plugins/cdti/include/cdti_searcher.hpp [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+bool cell_occupied(Point2i cell);
+private:
+  int cell_to_index(int x, int y, int width) { return y * width + x; }
+  Point2i index_to_cell(int index, int width) { return {index % width, index / width}; }
+
+
+
+  int width_;
+  int height_;
+  nav_msgs::msg::OccupancyGrid last_grid_;
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/cdti/include/cdti_searcher.hpp]: Replace OccupancyGrid with AGraph

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/tsp.cpp [hunk 14]
```diff
@@ -0,0 +1,181 @@
...
+#include <cdti.hpp>
+#include <utils.hpp>
+#include <pluginlib/class_list_macros.hpp>
+
+namespace cdti
+{
+void Plugin::initialize(as2::Node * node_ptr, std::shared_ptr<tf2_ros::Buffer> tf_buffer)
+{
+  node_ptr_ = node_ptr;
+  tf_buffer_ = tf_buffer;
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/tsp.cpp]: No need for tf_buffer

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/tsp.cpp [hunk 15]
```diff
@@ -0,0 +1,181 @@
...
+
+  cdti_routing_searcher_ = CDTIRoutingSearcher();
+  RCLCPP_INFO(node_ptr_->get_logger(), "Initializing cdti routing plugin");
+
+
+  // node_ptr_->declare_parameter("safety_distance", 0.5);
+  safety_distance_ = node_ptr_->get_parameter("safety_distance").as_double();
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/tsp.cpp]: No need for this since we're not using occupancy grids

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/tsp.cpp [hunk 16]
```diff
@@ -0,0 +1,181 @@
...
+  enable_path_optimizer_ = node_ptr_->get_parameter("enable_path_optimizer").as_bool();
+
+  // node_ptr_->declare_parameter("enable_visualization", true);
+  enable_visualization_ = node_ptr_->get_parameter("enable_visualization").as_bool();
+  penalty_x = node_ptr_->declare_parameter<double>("penalty_x", 1.0);
+  penalty_y = node_ptr_->declare_parameter<double>("penalty_y", 1.0);
+
+  enable_visualization_ = true;  // TODO(pariaspe): not publish when false
+
+  drone_mask_factor_ = node_ptr_->declare_parameter<int>("drone_mask_factor", 1);
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/tsp.cpp]: Also no need for this

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/tsp.cpp [hunk 17]
```diff
@@ -0,0 +1,181 @@
...
+      node_ptr_->create_publisher<visualization_msgs::msg::Marker>("plugin_viz/marker", 10);
+    viz_obstacle_grid_pub_ =
+      node_ptr_->create_publisher<nav_msgs::msg::OccupancyGrid>("plugin_viz/obstacle_map", 10);
+  }
+}
+
+void Plugin::occ_grid_cbk(const as2_msgs::msg::AGraph::SharedPtr msg)
+{
+  last_occ_grid_ = *(msg);
+}
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/tsp.cpp]: Replace with AGraph

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/tsp.cpp [hunk 18]
```diff
@@ -0,0 +1,181 @@
...
+  RCLCPP_INFO(node_ptr_->get_logger(), "Path size: %ld", path.size());
+
+  // Visualize path
+  auto path_marker = get_path_marker(
+    last_occ_grid_.header.frame_id, node_ptr_->get_clock()->now(), path,
+    last_occ_grid_.info, last_occ_grid_.header);
+  RCLCPP_INFO(node_ptr_->get_logger(), "Publishing path");
+  viz_pub_->publish(path_marker);
+
+  // TODO(pariasp): split path generator from visualization
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/tsp.cpp]: Don't leave TODOs for others

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/tsp.cpp [hunk 19]
```diff
@@ -0,0 +1,181 @@
...
+  std::vector<Point2i> path = cdti_routing_searcher_.solve_dijkstra(
+    last_occ_grid_, penalty_x, penalty_y);
+
+  if (path.size() == 0) {
+    RCLCPP_ERROR(node_ptr_->get_logger(), "Path to goal not found. Goal Rejected.");
+    return false;
+  }
+
+  if (enable_path_optimizer_) {
+    // TODO(pariaspe): Implement path optimizer
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/tsp.cpp]: Don't leave TODOs for other devs

File: as2_behaviors/as2_behaviors_path_planning/plugins/cdti/src/cdti_searcher.cpp [hunk 20]
```diff
@@ -0,0 +1,86 @@
...
+ *  \authors    Asil Arnous
+ ********************************************************************************/
+
+#include "cdti_searcher.hpp"
+#include <cmath>
+#include <fmt/ranges.h>
+#include <iostream>
+#include <string>
+
+std::vector<Point2i> CDTIRoutingSearcher::solve_dijkstra(
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/cdti/src/cdti_searcher.cpp]: Rename this method. You are not calculating Dijkstra's paths for all nodes in the graph

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/helperFunctions.cpp [hunk 21]
```diff
@@ -0,0 +1,262 @@
...
+        }
+
+        curr = predecessors[curr];
+    }
+
+    return {};
+}
+
+std::pair<std::vector<VertexDescriptor>, std::vector<double>>
+computeDijkstra(const Graph &g, int source)
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/helperFunctions.cpp]: Rename. It's weird to have solve_dijkstra and computeDijkstra in different namespaces

**Review by GPatiA2 — CHANGES_REQUESTED** [review]
Algorithm may work corretly, but input types should be corrected. Replace references of CDTI with TSP or something similar.

**Comment by alvgaona** [comment]
Hi @Asilon47 could please take a look at the rest of the PRs and adapt the title accordingly? Also, let's be careful with the merging strategy because there are a lot of non-descriptive commit messages. 

cc @GPatiA2

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/tsp_searcher.cpp [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+    for (int idx : path) {
+        point_path.emplace_back(static_cast<int>(x_vec[idx]), static_cast<int>(y_vec[idx]));
+    }
+
+
+    return point_path;
+}
+bool TSPRoutingSearcher::cell_occupied(Point2i cell) {
+    return false;  
+}
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/tsp_searcher.cpp:83]: This is not necessary. Don't ask if the cell is occupied if you always return false

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/helperFunctions.h [hunk 23]
```diff
@@ -0,0 +1,49 @@
...
+std::vector<int> getPath(const std::vector<VertexDescriptor>& predecessors, int target);
+
+Graph buildGraphFromRaw(
+    const double *x_coords,
+    const double *y_coords,
+    const std::vector<std::vector<int>> &adj_list,
+    double penalty_x, double penalty_y);
+std::vector<int> greedyTargetTSP(const Graph& graph, int startVertex, const std::vector<int>& targets);
+void sortVertices(std::vector<std::pair<double, double>>& points);
+bool edgeIntersectsPolygon(const std::pair<double, double>& u, const std::pair<double, double>& v, const std::vector<std::pair<double, double>>& polygon);
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/helperFunctions.h]: What are these functions for? Your plugin only receives a graph with no obstacles

File: as2_msgs/msg/AGraph.msg [hunk 24]
```diff
@@ -0,0 +1,7 @@
+std_msgs/Header header
+nav_msgs/MapMetaData info
+NProperties[] nproperties
+Vertecies[] adjesency_list
+uint32 start_node
+Vertecies[] route
```
Comment (GPatiA2) [inline as2_msgs/msg/AGraph.msg:6]: What are you using this field for? Isn't there already a path message that does the job?

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/CMakeLists.txt [hunk 25]
```diff
@@ -0,0 +1,72 @@
...
+install(
+  TARGETS ${PLUGIN_NAME}
+  EXPORT export_${PLUGIN_NAME}
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin
+)
+
+# PLUGIN TESTS
+# if(BUILD_TESTING)
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/CMakeLists.txt]: Enable testing to check code style.

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/tsp.cpp [hunk 26]
```diff
@@ -0,0 +1,174 @@
...
+
+  tsp_routing_searcher_ = TSPRoutingSearcher();
+  RCLCPP_INFO(node_ptr_->get_logger(), "Initializing tsp routing plugin");
+
+
+
+/*
+  drone_mask_factor_ = node_ptr_->get_parameter("drone_mask_factor").as_int();
+*/
+  // node_ptr_->declare_parameter("enable_path_optimizer", false);
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/tsp.cpp]: Remove comments

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/helperFunctions.cpp [hunk 27]
```diff
@@ -0,0 +1,262 @@
...
+std::pair<double, double> getCentroid(const std::vector<std::pair<double, double>> &points)
+{
+    double sx = 0.0, sy = 0.0;
+    for (const auto &p : points)
+    {
+        sx += p.first;
+        sy += p.second;
+    }
+    return {sx / points.size(), sy / points.size()};
+}
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/helperFunctions.cpp]: Remove this function, it's not used anywhere

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/helperFunctions.cpp [hunk 28]
```diff
@@ -0,0 +1,262 @@
...
+{
+    for (const auto &poly : polygons)
+    {
+        if (edgeIntersectsPolygon(u, v, poly))
+        {
+            return true;
+        }
+    }
+    return false;
+}
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/helperFunctions.cpp]: Remove these functions. Plugin receives a graph with no obstacles, so there is no need to do what these functions do. Also, these are not used anywhere

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/tsp.cpp [hunk 29]
```diff
@@ -0,0 +1,174 @@
...
+
+visualization_msgs::msg::Marker Plugin::get_path_marker(
+  std::string frame_id, rclcpp::Time stamp,
+  std::vector<Point2i> path, nav_msgs::msg::MapMetaData map_info,
+  std_msgs::msg::Header map_header)
+{
+  visualization_msgs::msg::Marker marker;
+  marker.header.frame_id = frame_id;
+  marker.header.stamp = stamp;
+  marker.ns = "a_star";
```
Comment (GPatiA2) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/src/tsp.cpp]: Change namespace to something like TSP

**Review by GPatiA2 — CHANGES_REQUESTED** [review]
Minor changes:

- Remove unused code
- Remove non-informative commentaries
- Enable testing so you can fix code style
- Check for the usage of the `route` field in AGraph message

File: as2_msgs/msg/AGraph.msg (hunk 24 above)
Comment (Asilon47) [reply as2_msgs/msg/AGraph.msg:6]: it is needed to specify nodes to pass inside the route

**Review by GPatiA2 — CHANGES_REQUESTED** [review]
Remove the gate_texture from the PR so it doesn't mess up with

**Review by GPatiA2 — APPROVED** [review]

**Comment by GPatiA2** [comment]
PR seems good to me now (as long as all tests are passed). But I've got some questions left:

- This PR includes a dependency in Aerostack2, the Boost graph library.
- This PR also includes new messages in as2_messages that are directly coupled with the chosen Boost data structure to represent graphs.

Is this okay? Should the plugin be implemented without relying on external libraries?

**Comment by GPatiA2** [comment]
Feedback is welcome about this @pariaspe @RPS98 @miferco97

File: as2_msgs/msg/Node.msg
Comment (RPS98) [inline as2_msgs/msg/Node.msg:1]: Maybe use a more specific name for msgs, but generic across planners:
- Graph 
- GraphNode

File: as2_msgs/msg/Vertecies.msg
Comment (RPS98) [inline as2_msgs/msg/Vertecies.msg:1]: What is this message?

File: as2_simulation_assets/as2_gazebo_assets/models/cvar_gate/meshes/cvar_gate_texture.png
Comment (RPS98) [inline as2_simulation_assets/as2_gazebo_assets/models/cvar_gate/meshes/cvar_gate_texture.png:1]: Do not remove this, add the same image as main has

File: as2_behaviors/as2_behaviors_path_planning/plugins.xml [hunk 30]
```diff
@@ -9,4 +9,9 @@
       <description>This is Voronoi plugin.</description>
     </class>
   </library>
+  <library path="tsp">
+    <class type="tsp::Plugin" base_class_type="as2_behaviors_path_planning::PluginBase">
+      <description>This is TSP routing plugin.</description>
```
Comment (RPS98) [inline as2_behaviors/as2_behaviors_path_planning/plugins.xml:14]: Improve the description. How it solves the TSP problem

**Review by RPS98 — CHANGES_REQUESTED** [review]

File: as2_msgs/msg/Node.msg
Comment (pariaspe) [reply as2_msgs/msg/Node.msg:1]: Agree. A few extra things:
- Correct typos: adjesency_list -> adjacency_list, verticeis -> vertices.
- I suggest to change name `NProperties` to `Node`.
- Also, not sure if `start_node` and `route` should be encoded in the same Graph msg. 
- Why is MapMetaData needed?

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/helper_functions.h
Comment (pariaspe) [inline as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/helper_functions.h:1]: Please follow Aerostack2 (and ROS 2) style cpp [guidelines](https://docs.ros.org/en/rolling/The-ROS2-Project/Contributing/Code-Style-Language-Versions.html):
- Files should be in `snake_case`
- Boost should be avoided unless absolutely required
- `Pragma directive should be also avoided

File: as2_msgs/msg/Node.msg
Comment (Asilon47) [reply as2_msgs/msg/Node.msg:1]: changed the names as requested and corrected typos, MapMetaData is for visualization

File: as2_msgs/msg/Vertecies.msg
Comment (Asilon47) [reply as2_msgs/msg/Vertecies.msg:1]: changed its name into Uint32MultiArray, it is simply a vector of vectors

File: as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/helper_functions.h
Comment (Asilon47) [reply as2_behaviors/as2_behaviors_path_planning/plugins/tsp/include/helper_functions.h:1]: removed boost, changed files name, and removed #pragma once

File: as2_behaviors/as2_behaviors_path_planning/plugins.xml (hunk 30 above)
Comment (GPatiA2) [reply as2_behaviors/as2_behaviors_path_planning/plugins.xml:14]: @Asilon47 address this! It shouldn't take long

File: as2_msgs/msg/Node.msg
Comment (pariaspe) [reply as2_msgs/msg/Node.msg:1]: Still missing the `vertecies` one. Please, change it to `vertices`
